custom_components/econnect_metronet/coordinator.py

```diff
@@ -0,0 +1,96 @@
+import logging
+from datetime import timedelta
+from typing import Any, Dict, Optional
+
+import async_timeout
+from elmo.api.client import ElmoClient
+from elmo.api.exceptions import InvalidToken
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    CONF_DOMAIN,
+    CONF_SCAN_INTERVAL,
+    CONF_SYSTEM_URL,
+    DOMAIN,
+    POLLING_TIMEOUT,
+    SCAN_INTERVAL_DEFAULT,
+)
+from .devices import AlarmDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AlarmCoordinator(DataUpdateCoordinator):
+    def __init__(self, hass: HomeAssistant, config: ConfigEntry) -> None:
+        # Initialize all clients and devices
+        client = ElmoClient(config.data[CONF_SYSTEM_URL], config.data[CONF_DOMAIN])
+        scan_interval = config.options.get(CONF_SCAN_INTERVAL, SCAN_INTERVAL_DEFAULT)
+
+        # Attributes
+        self.device = AlarmDevice(client, config.options)
```

Comment:
Suggestion: maybe it's better if the `AlarmDevice` is built externally and passed as a constructor parameter. The coordinator, coordinates the update and doesn't care much about the device itself.